### ISSUE TITLE
TypeError: Object #<Socket> has no method 'uncork'

### ISSUE THREAD

**Issue by mrvisser** [NodeRedis/node_redis#897]
We're currently seeing this error in a production server:

```
TypeError: Object #<Socket> has no method 'uncork'
    at Multi.exec_transaction (/opt/oae/node_modules/redis/index.js:1105:25)
    ...
```

The invocation that leads to this enters the Redis module [here](https://github.com/oaeproject/Hilary/blob/12.1.1/node_modules/oae-activity/lib/internal/dao.js#L502).

Our RedisClient initialization logic happens [here](https://github.com/oaeproject/Hilary/blob/12.1.1/node_modules/oae-util/lib/redis.js#L47).

Node 0.10.17
node_redis: 2.2.5

Please let me know if there's more I can do to help!

**Comment by BridgeAR** [comment]
Using Node 0.12.x would solve the issue for now. I'll have a closer look at it later on why the fallback mode does not work properly in your case. I guess I already know why it's not working for you and if that's the case I'll release 2.2.6 later on today.

**Comment by mrvisser** [comment]
Thanks @BridgeAR , much appreciated!

**Comment by BridgeAR** [comment]
@mrvisser I'm sorry I did not find the time to fix it today.

**Comment by mrvisser** [comment]
@BridgeAR no worries. If you like, I can diagnose and try and get a PR in. If you could share what your hunch is on the issue then that might help. Let me know.

**Comment by BridgeAR** [comment]
@mrvisser I had another look at it and I can't reproduce the issue on 0.10.27 but I guess the stream module might somehow got altered by something else. Nevertheless I changed and improved the way how the fallback mode works and that should also fix your issue.

I got the fix local and I'm going to push that and some other things later on today. It would be great if you could test the fix before releasing a new version :)

**Comment by mrvisser** [comment]
Thanks, I'll try and reproduce today with a unit test as part of our issue that references this one and verify it resolves it.

Cheers!

**Comment by BridgeAR** [comment]
@mrvisser would you be so kind and give my fix a try?

**Comment by mrvisser** [comment]
@BridgeAR today I've tried to write a unit test in my application that reproduces this issue, however I can't seem to trigger it. I've tried restricting bandwidth and increasing the size of the multi commands to produce back-pressure, but it doesn't seem to trigger the issue.

I think I'll just have to upgrade the client version and deploy it, which would have to be a bit later this week. I'll get back when we've done this.

**Comment by BridgeAR** [comment]
@mrvisser the issue occurs after a connection loss. It is indeed a bug and I fixed it in #901

**Comment by mrvisser** [comment]
@BridgeAR I've pulled your "stuff" branch into my local instance and run the unit tests and can confirm there aren't any regressions in that branch. If released to npm, we'd be happy to give a 2.2.6 a try.

**Comment by shezarkhani** [comment]
A release for this would be great. We're seeing a lot of these errors too.
